src/main/java/Thea.java

```diff
@@ -0,0 +1,181 @@
+                    }
+                } else {
+                    throw new WrongCommandException("Sorry, I don't understand what that means.. '^'");
+                }
+            }
+            saveTaskList(tasks);
+            userInput = input.nextLine();
+        }
+    }
+
+    public static void saveTaskList(ArrayList<Task> tasks) {
+        String currentDir = System.getProperty("user.dir");
+        Path dataDirPath =Paths.get(currentDir, "data");
+        Path path = Paths.get(currentDir, "data", "thea.txt");
+        if(!Files.exists(dataDirPath)) {
+            try {
+                Files.createDirectories(dataDirPath);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        } else {
+            if (!Files.exists(path)) {
+                try {
+                    Files.createFile(path);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+        try (BufferedWriter bufferWriter = Files.newBufferedWriter(path)) {
+            for (Task task: tasks) {
+                bufferWriter.write(task.toMemoryFormat());
+                bufferWriter.newLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static ArrayList<Task> retrieveTasks() throws Exception{
+        String currentDir = System.getProperty("user.dir");
+        Path dataDirPath =Paths.get(currentDir, "data");
+        Path path = Paths.get(currentDir, "data", "thea.txt");
+        if(!Files.exists(dataDirPath)) {
+            return new ArrayList<>();
+        }
+        if (!Files.exists(path)) {
+            return new ArrayList<>();
+        }
+        String line;
+        ArrayList<Task> retrievedTasks = new ArrayList<>();
+        Task task;
+        try (BufferedReader bufferReader = Files.newBufferedReader(path)) {
+            line = bufferReader.readLine();
+            while (line != null) {
+                String[] splitLine = line.split(" [|] ");
+                if (splitLine[0].equals("T")) {
+                    task = new ToDo(splitLine[2]);
+                } else if (splitLine[0].equals("D")) {
+                    task = new Deadline(splitLine[2], splitLine[3]);
+                } else if (splitLine[0].equals("E")) {
+                    task = new Event(splitLine[2],splitLine[3], splitLine[4]);
+                } else {
+                    throw new FileCorruptedException("Unexpected File Format Found. File might be corrupted.");
+                }
+                if (splitLine[1].equals("1")) {
+                    task.markAsDone();
+                }
+                retrievedTasks.add(task);
+                line = bufferReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);

```

Comment:
Do consider abstracting out the parsing into a separate function to avoid "arrowhead" style code!